eth/polar/mining.go

```diff
@@ -30,7 +30,13 @@ import (
 // TODO: replace this file with a proper mining object and use message passing instead of direct calls.
 // Prepare prepares the Polaris chain for processing a new block at the given height.
 func (pl *Polaris) Prepare(ctx context.Context, number uint64) {
-	pl.blockchain.Prepare(ctx, number)
+	header := pl.blockchain.Prepare(ctx, number)
+	if header == nil {
+		panic("blockchain produced nil header")
+	}
+	// We update the base fee in the txpool to the next base fee.
+	// TODO: Move to prepare proposal
+	pl.txPool.SetBaseFee(header.BaseFee)
```

Comment:
this should happen in the previous block's endblock/precommit

Reply:
that will break consensus :(

Reply:
needs to be done in process / prepare proposal to be 100% correct. I think lets push off changing anything until I get there.

Reply:
also probs gonna be addressed in #1107